### DISCUSSION TITLE
[错误报告]: 订阅规则的疑惑

### DISCUSSION THREAD

**Discussion by Whichbfj28** [jxxghp/MoviePilot#876] [讨论]
### 确认

- [X] 我的版本是最新版本，我的版本号与 [version](https://github.com/jxxghp/MoviePilot/releases/latest) 相同。
- [X] 我已经 [issue](https://github.com/jxxghp/MoviePilot/issues) 中搜索过，确认我的问题没有被提出过。
- [X] 我已经 [Telegram频道](https://t.me/moviepilot_channel) 中搜索过，确认我的问题没有被提出过。
- [X] 我已经修改标题，将标题中的 描述 替换为我遇到的问题。

### 当前程序版本

1.3.3

### 问题类型

主程序运行问题

### 问题描述

订阅规则如下：
![image](https://github.com/jxxghp/MoviePilot/assets/47915665/0ed422a3-b042-4cfd-abfc-5c7df12094a0)

电影标题标题识别如下
鬼吹灯之南海归墟.South.Sea.Tomb.2022.1080p.WEB-DL.H264.AAC
![image](https://github.com/jxxghp/MoviePilot/assets/47915665/60f6b01d-aa79-41a5-8a07-09fdbcbcf1a8)

问题来了！
没有命中任何订阅规则。这是我理解错了。还是程序问题：
![image](https://github.com/jxxghp/MoviePilot/assets/47915665/3c6d2e78-692b-47a4-9e0a-8c0bb89eb8bf)

**Comment by dlovew** [comment]
你这规则确实不会命中，4k和1080p不会同时生效

**Comment by Whichbfj28** [comment]
> 你这规则确实不会命中，4k和1080p不会同时生效

没看明白。这种子可以识别为1080p。也不符合默认排除的条件。应该命中优先级2才是。为何不命中？
不会同时生效的话。哪怕生效1080p也可以命中才是
我同时选择1080p、4K是表示。这2个分辨率我都接受的意思

**Comment by dlovew** [comment]
> > 你这规则确实不会命中，4k和1080p不会同时生效
> 
> 没看明白。这种子可以识别为1080p。也不符合默认排除的条件。应该命中优先级2才是。为何不命中？ 不会同时生效的话。哪怕生效1080p也可以命中才是 我同时选择1080p、4K是表示。这2个分辨率我都接受的意思

你的规则1要求h265但是你的标题并不带，规则2本身就错了，不能同时选两种分辨率，你以为两种都选是两种都接受，但是程序不是这样设定的，也不会因为你一个人改变设定，如果4k，1080p都能接受，就设置规则3

**Comment by Whichbfj28** [comment]
> > > 你这规则确实不会命中，4k和1080p不会同时生效
> > 
> > 
> > 没看明白。这种子可以识别为1080p。也不符合默认排除的条件。应该命中优先级2才是。为何不命中？ 不会同时生效的话。哪怕生效1080p也可以命中才是 我同时选择1080p、4K是表示。这2个分辨率我都接受的意思
> 
> 你的规则1要求h265但是你的标题并不带，规则2本身就错了，不能同时选两种分辨率，你以为两种都选是两种都接受，但是程序不是这样设定的，也不会因为你一个人改变设定，如果4k，1080p都能接受，就设置规则3

规则2。相信不是我一个人这样理解的吧。这是多选框。勾选上的规则代表我接受这个的意思
如果程序如此设计的话。应该在选择1080P的时候。把4K变灰不让选择。

**Comment by dlovew** [comment]
> > > > 你这规则确实不会命中，4k和1080p不会同时生效
> > > 
> > > 
> > > 没看明白。这种子可以识别为1080p。也不符合默认排除的条件。应该命中优先级2才是。为何不命中？ 不会同时生效的话。哪怕生效1080p也可以命中才是 我同时选择1080p、4K是表示。这2个分辨率我都接受的意思
> > 
> > 
> > 你的规则1要求h265但是你的标题并不带，规则2本身就错了，不能同时选两种分辨率，你以为两种都选是两种都接受，但是程序不是这样设定的，也不会因为你一个人改变设定，如果4k，1080p都能接受，就设置规则3
> 
> 规则2。相信不是我一个人这样理解的吧。这是多选框。勾选上的规则代表我接受这个的意思 如果程序如此设计的话。应该在选择1080P的时候。把4K变灰不让选择。

一开始确实有你这样理解的，但是tg群早就说过了，其实你可以先去tg群问一下，里面很多热心人，解决不了的再提issue，窝着也只是个建议。另外对于选了1080p,4k不会变灰色的问题，可能是作者忽略了，也可能难实现，毕竟一开始作者就说过他前端不是特别好，而且如果有特殊诉求可以自己提pr

**Comment by stanlli** [comment]
求教正确的订阅规则应该是什么样的呢？
